knpy/braid.py

```diff
@@ -188,25 +188,21 @@ def destabilization(self,inplace = True):
         else:
             raise IllegalTransformationException
     
-    def remove_sigma_inverse_pair(self,index,inplace=True):
+    def remove_sigma_inverse_pair(self,index):
         """
-        Remove consequtive inverse on a given place
+        Remove consequtive inverse on a given place (last and first element are consequtive)
+        index: the smaller index (or when the pair is the last and first then it is the last index)
         """
         if self.is_remove_sigma_inverse_pair_performable(index):
-            transformed_braid = self._braid
-            mask = np.ones(self._braid.shape,dtype="bool")
-            mask[index] = False
-            mask[(index+1)%self._braid.shape[0]] = False
-            transformed_braid = transformed_braid[mask]
-
-            if inplace:
-                self._braid = transformed_braid
-                if(self._braid.shape[0] == 0):
-                    self._n = 1
-                else:
-                    self._n = np.max(np.abs(self._braid)) + 1
+            if index == 0:
+                modified_braid = self._braid[2:]
+            elif index == self._braid.shape[0] - 2:
+                modified_braid = self._braid[:-2]
+            elif index == self._braid.shape[0] - 1:
+                modified_braid = self._braid[1:-1]
             else:
-                return transformed_braid
+                modified_braid = np.concatenate((self._braid[:(index - 1)], self._braid[(index + 1):]))
+            return modified_braid
```

Comment:
Method should return a new `Braid` object, this returns a Numpy array as it is now.